scripts/bug_retriever.py

```diff
@@ -60,7 +60,15 @@ def retrieve_bugs(self):
 
         # Try to re-download inconsistent bugs, up to three times.
         for i in range(3):
-            bug_ids = set(bug_snapshot.get_inconsistencies())
+            # We look for inconsistencies in all bugs first, then, on following passes,
+            # we only look for inconsistencies in bugs that were found to be inconsistent in the first pass
+            if i == 0:
+                inconsistent_bugs_set = bugzilla.get_bugs()
+            else:
+                inconsistent_bugs_set = inconsistent_bugs
+            inconsistent_bugs = bug_snapshot.get_inconsistencies(inconsistent_bugs_set)
```

Comment:
I don't think this is going to work. On the first pass, you are passing an iterator on bugs, on the second you are passing a list of bug IDs (instead, it should be a list of bugs).

You need to modify `get_inconsistencies` to return bugs instead of just their IDs.